web/src/components/core/Field.jsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) [2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+// @ts-check
+
+import React from "react";
+import { Icon } from "~/components/layout";
+import { If } from "~/components/core";
+
+/**
+ * @typedef {import("~/components/layout/Icon").IconName} IconName
+ * @typedef {import("~/components/layout/Icon").IconSize} IconSize
+ */
+
+/**
+ * @typedef {object} FieldProps
+ * @property {React.ReactNode} label - The field label.
+ * @property {React.ReactNode} [value] - The field value.
+ * @property {React.ReactNode} [description] - A field description, useful for providing context to the user.
+ * @property {IconName} [icon] - The name of the icon for the field.
+ * @property {IconSize} [iconSize] - The size for the field icon.
+ * @property {string} [className] - ClassName
+ * @property {() => {}} [onClick] - Callback
+ * @property {React.ReactNode} [children] - A content to be rendered as field children
+ *
+ * @typedef {Omit<FieldProps, 'icon'>} FieldPropsWithoutIcon
+ */
+
+/**
+ * Component for laying out a page field
+ *
+ * @param {FieldProps} props
+ */
+const Field = ({
```

Comment:
I've called it _Field_ but definitely is not the best name. A field usually has a label and value, which is the case for most of the use cases it will have. It might happen that a field does not have a _simple value_ but a complex one, which is the case of the "Partitions and file systems" (see attached screenshot in the PR's description). In this case, the value is actually rendered in the _content_ part, a.k.a _children_. 

So, this looks like more a _subsection_ or so, but I'm not sure. Anyway, if you have a better naming in mind... please speak up. If not, never will be late for running the [sed](https://linux.die.net/man/1/sed) command against the whole code base :stuck_out_tongue_winking_eye:

Reply:
"Field" is fine for now.